var/spack/repos/builtin/packages/py-tensorflow/package.py

```diff
@@ -141,7 +141,8 @@ class PyTensorflow(Package, CudaPackage):
     depends_on('py-keras-preprocessing@1.1.0:', type=('build', 'run'), when='@2.1:')
     depends_on('py-keras-preprocessing@1.0.5:', type=('build', 'run'), when='@1.12:')
     depends_on('py-keras-preprocessing@1.0.3:', type=('build', 'run'), when='@1.11:')
-    depends_on('py-numpy@1.16.0:1.999',  type=('build', 'run'), when='@1.13.2,1.15:')
+    depends_on('py-numpy@1.18.5',        type=('build', 'run'), when='@2.0.0:')
+    depends_on('py-numpy@1.16.0:1.999',  type=('build', 'run'), when='@1.13.2,1.15')
     depends_on('py-numpy@1.14.5:1.999',  type=('build', 'run'), when='@1.12.1,1.14.0')
```

Comment:
```suggestion
    # https://github.com/tensorflow/tensorflow/issues/40688
    depends_on('py-numpy@1.16.0:1.18',  type=('build', 'run'), when='@1.13.2,1.15:')
    depends_on('py-numpy@1.14.5:1.18',  type=('build', 'run'), when='@1.12.1,1.14.0')
```
I think this will be more robust. If 1.19 has an API breaking change, I doubt older versions can handle it either.

Reply:
Thank you for your advice.
I fixed according to your advice.